### ISSUE TITLE
Missing documentation on installing Mkdocs development version dependencies

### ISSUE THREAD

**Issue by pcgaustad** [mkdocs/mkdocs#3025]
I can't find any information in the documentation about installing the necessary dependencies for Mkdocs.

With Python 3.10.6 in a fairly standard Ubuntu 22.04 WSL environment I had to install the following dependencies after running `pip install --editable .` to be able to run `mkdocs serve`:

```bash
pip install markdown pymdown-extensions markdown-callouts mdx-gh-links \
    mkdocs-{redirects,click,autorefs,literate-nav} mkdocstrings[python]
```

Is there a more obvious way? `hatch init --new` doesn't work in my case, at least.

If there is a way that would work across different environments, I'd be happy to add it to the documentation.

**Comment by ofek** [comment]
`hatch env docs` isn't a command

**Comment by oprypin** [comment]
You're right that this isn't documented in our repo :confused: 

---

To build docs (installs deps implicitly):
`hatch run docs:mkdocs serve`

To learn about the dependencies:
`hatch env show`
`hatch env show docs`

<details><summary>Click to expand<br><code>hatch env show --json | jq -r ".docs.dependencies | .[]"</code></summary>

```
Markdown >=3.3.3
mdx_gh_links >=0.2
markdown-callouts >=0.3.0
mkdocs-literate-nav >=0.5.0
mkdocs-redirects >=1.0.1
pymdown-extensions >=8.0.1
mkdocstrings-python >=0.7.1
mkdocs-click >=0.8.0
```

</details>

To roll with your own dependency installation method:
`hatch env show --json | jq -r ".docs.dependencies | .[]" > docs.requirements.txt`
`pip install -r docs-requirements.txt`

> (cc @ofek I really wish this didn't require `jq`)

**Comment by ofek** [comment]
Could do https://github.com/repo-helper/hatch-requirements-txt + https://hatch.pypa.io/latest/config/environment/overview/#features

**Comment by oprypin** [comment]
No, I want to go the other way, print out a requirements.txt on the fly, not store a requirements.txt file

**Comment by ofek** [comment]
You could forget the first part then and do https://hatch.pypa.io/latest/cli/reference/#hatch-dep-show-requirements

**Comment by oprypin** [comment]
Thanks. It works to show the dependencies of the main project, but I couldn't figure out how to make it show dependencies of the 'docs' env.

**Comment by ofek** [comment]
Right I'm saying you could make those a feature, which might be more of a thing soon https://discuss.python.org/t/adding-a-non-metadata-installer-only-dev-dependencies-table-to-pyproject-toml/20106/45

**Comment by ofek** [comment]
> show dependencies of the 'docs' env

Please open a FR

**Comment by pcgaustad** [comment]
> To build docs (installs deps implicitly):
> `hatch run docs:mkdocs serve`

Thanks! Another way would be `hatch -e docs shell`, it seems,

**Comment by joho1968** [comment]
Perhaps I'm stating the obvious here, but this applies to those who want to contribute to the translation efforts. (I just ran into this while creating a Swedish translation.)
